pkg/api/controller.go

```diff
@@ -1731,6 +1731,134 @@ func (c *Controller) GetRepositoryMetadata(w http.ResponseWriter, r *http.Reques
 	writeResponse(w, r, http.StatusOK, apigen.RepositoryMetadata{AdditionalProperties: metadata})
 }
 
+func (c *Controller) GetBranchProtectionRules(w http.ResponseWriter, r *http.Request, repository string) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.GetBranchProtectionRulesAction,
+			Resource: permissions.RepoArn(repository),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	rules, eTag, err := c.Catalog.GetBranchProtectionRules(ctx, repository)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	resp := make([]*apigen.BranchProtectionRule, 0, len(rules.BranchPatternToBlockedActions))
+	for pattern := range rules.BranchPatternToBlockedActions {
+		resp = append(resp, &apigen.BranchProtectionRule{
+			Pattern: pattern,
+		})
+	}
+	if eTag == "" {
+		// workaround since swagger doesn't allow empty headers
+		// https://github.com/deepmap/oapi-codegen/issues/954
+		eTag = base64.StdEncoding.EncodeToString([]byte("EMPTY"))
+	}
+	w.Header().Set("ETag", eTag)
+	writeResponse(w, r, http.StatusOK, resp)
+}
+
+func (c *Controller) SetBranchProtectionRules(w http.ResponseWriter, r *http.Request, body apigen.SetBranchProtectionRulesJSONRequestBody, repository string, params apigen.SetBranchProtectionRulesParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.SetBranchProtectionRulesAction,
+			Resource: permissions.RepoArn(repository),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "create_branch_protection_rule", r, repository, "", "")
+
+	// For now, all protected branches use the same default set of blocked actions. In the future this set will be user configurable.
+	blockedActions := []graveler.BranchProtectionBlockedAction{graveler.BranchProtectionBlockedAction_STAGING_WRITE, graveler.BranchProtectionBlockedAction_COMMIT}
+
+	rules := &graveler.BranchProtectionRules{
+		BranchPatternToBlockedActions: make(map[string]*graveler.BranchProtectionBlockedActions),
+	}
+	for _, r := range body {
+		rules.BranchPatternToBlockedActions[r.Pattern] = &graveler.BranchProtectionBlockedActions{
+			Value: blockedActions,
+		}
+	}
+	eTag := params.IfMatch
+	if swag.StringValue(eTag) == "" {
+		// workaround since swagger doesn't allow empty headers
+		// https://github.com/deepmap/oapi-codegen/issues/954
+		eTag = swag.String(base64.StdEncoding.EncodeToString([]byte("EMPTY")))
+	}
```

Comment:
I don't understand this.  If there is no `If-Match` header, I think this makes it behave as though we _did_ ask for a particular ETag.

Reply:
You're right, this is not needed.